modules/upload/upload.service.js

```diff
@@ -0,0 +1,119 @@
+                    return resolve(fileBuffer);
+                });
+
+                inputStream.on('data', (chunk) => {
+                    chunks.push(chunk);
+                });
+            })
+
+            Jimp.read(inputBuffer).then(image => {
+                image.resize(1920, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
+                    const
+                        blobName = `large_${id}_${filename}`,
+                        stream = getStream(buffer),
+                        streamLength = buffer.length
+
+
+                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
+                        if (err) {
+                            throw new Error(err)
+                        }
+                    })
+                })
+                image.resize(1080, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
+                    const
+                        blobName = `medium_${id}_${filename}`,
+                        stream = getStream(buffer),
+                        streamLength = buffer.length
+
+                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
+                        if (err) {
+                            throw new Error(err)
+                        }
+                    })
+                })
+
+                image.resize(768, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
+                    const
+                        blobName = `small_${id}_${filename}`,
+                        stream = getStream(buffer),
+                        streamLength = buffer.length
+
+                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
+                        if (err) {
+                            throw new Error(err)
+                        }
+                    })
+                })
+
+                image.resize(128, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
+                    const
+                        blobName = `thumbnail_${id}_${filename}`,
+                        stream = getStream(buffer),
+                        streamLength = buffer.length
+
+                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
+                        if (err) {
+                            throw new Error(err)
+                        }
+                    })
+                })
+            })
+
+            return {
+                prefixUrl: process.env.IMAGE_LINK,
+                fileNames: {
+                    large: `large_${id}_${filename}`,
+                    medium: `medium_${id}_${filename}`,
+                    small: `small_${id}_${filename}`,
+                    thumbnail: `thumbnail_${id}_${filename}`,
+                }
+            }
+        })
+    }
+
+    async deleteFiles(files) {
+        return files.map(fileName => {
```

Comment:
Please, use here `Promise.all` to return response only when all files are deleted